# Тести для малих вибірок та пропорцій



**Завдання 1**. E-commerce компанія після редизайну сайту підозрює, що **середній час до покупки (time-to-purchase)** користувачів **збільшився**.
Історично середній час від першого заходу на сайт до покупки становив $\mu_0 = 123.7$ хвилин.

Команда аналітиків випадково обрала дані **7 покупців після редизайну**:
`128, 135, 121, 142, 126, 151, 123`

З допомогою наявних даних зʼясуйте, чи збільшився середній час покупки після редизайну.

Для цього виконайте наступні 6 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть нульову та альтернативну гіпотези і визначте тип тесту.
2. Обчисліть вибіркові статистики: $\bar x$, $s$, $n$.
3. Оберіть тип тесту та виконайте його (знайдіть критичне значення тесту, статистику тесту та р-значення) будь-яким пасуючим способом, щоб перевірити гіпотезу на рівні значущості $\alpha = 0.10$.
4. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$ використовуючи p-value.
5. Напишіть висновок: чи справді редизайн сайту подовжив час до покупки?
6. Чи зміниться ваше рішення при зміні рівня значущості на $\alpha = 0.05$.


Ho = 123.7

Ha > 123,7

тест правосторонній

In [1]:
import numpy as np
from scipy import stats

In [3]:
redesing = np.array([128, 135, 121, 142, 126, 151, 123])
mu0 = 123.7


In [4]:
x_avg = np.mean(redesing)
s = np.std (redesing, ddof=1)
n = len(redesing)


In [6]:
print ('Вибіркове середнє = ',x_avg.round(2))
print ('Вибіркове стандартне відхилення = ',s.round(2))
print ('Кількість значень ',n)

Вибіркове середнє =  132.29
Вибіркове стандартне відхилення =  10.98
Кількість значень  7


In [7]:
alpha = 0.1

In [12]:
# scipy тест
calc = stats.ttest_1samp(redesing, popmean=mu0, alternative="greater")
print ('scipy тест')
print ('t - статистика =', calc.statistic.round(4))
print ('p-value =', calc.pvalue.round(4))

scipy тест
t - статистика = 2.0687
p-value = 0.042


In [14]:
# висновок на основі р-значення
if calc.pvalue < alpha:
    print(f"Відхиляємо Ho: середній час на редізайн збільшився на рівні значущості {alpha}")
else:
    print("Не відхиляємо H0: доказів недостатньо")

Відхиляємо Ho: середній час на редізайн збільшився на рівні значущості 0.1


In [15]:
alpha1 = 0.05


In [16]:
# висновок на основі р-значення
if calc.pvalue < alpha1:
    print(f"Відхиляємо Ho: середній час на редізайн збільшився на рівні значущості {alpha1}")
else:
    print("Не відхиляємо H0: доказів недостатньо")

Відхиляємо Ho: середній час на редізайн збільшився на рівні значущості 0.05


і при рівні значущості альфа 0,1, і при альфа 0,05 значення p-value (0,042) менше, тобто дійсно середній час до покупки збільшився

**Завдання 2.**

До спеціальної рекламної кампанії **23%** дорослих упізнавали логотип компанії. Після завершення кампанії відділ маркетингу провів опитування: з **1200** випадково відібраних дорослих **311** упізнали логотип.

Перевірте на рівні значущості **3%** ($\alpha=0.03$), чи дають ці дані достатні підстави стверджувати, що **тепер більше ніж 23%** дорослих упізнають логотип компанії. Для розвʼязку використовуйте бібліотеку `statsmodels`.

Зробіть висновок, чи зросла впізнаваність логотипу.

Додатково, обчисліть довірчий інтревал на заданому рівні значущості і проінтерпретуйте текстом - як він додатково пояснює прийняте нами рішення?

In [17]:
import statsmodels.api as sm
from statsmodels.stats.proportion import proportions_ztest, proportion_confint

In [34]:
count = 311
nobs = 1200
p0 = 0.23
alpha_task2 = 0.03

# z-тест для пропорції
z_stat, p_val = proportions_ztest(count, nobs, p0, alternative='larger')

In [31]:
print("Тест для однієї пропорції")
print('Z-статистика = ', z_stat.round(4))
print('p-value = ',p_val.round(4))

Тест для однієї пропорції
Z-статистика =  2.3058
p-value =  0.0106


Ho = 23%

Ha > 23%

рівень значущості 0,03

Оскільки  p-value=0.0106 < 0.03 , відхиляємо  Ho . Є статистично значущі докази, що після проведення маркетингової кампанії, логотип компанії узнають більше, ніж 23% людей

In [40]:
confint = proportion_confint(count, nobs, alpha=alpha_task2, method='normal')
print(f"97% довірчий інтервал: {confint}")

97% довірчий інтервал: (0.23171700302179205, 0.28661633031154127)


97% довірчий інтервал (0.2317, 0.0,2866)

Ніжня межа довірчого інтервалу 23,17%, що підтверджує, що рекламна кампанія принесла результати і логотип пізнають більше ніж 23% людей
